pkg/loop/internal/pb/relayer.proto

```diff
@@ -0,0 +1,306 @@
+  rpc NodeStatuses (NodeStatusesRequest) returns (NodeStatusesReply) {}
+
+  rpc SendTx (SendTxRequest) returns (google.protobuf.Empty) {}
+}
+
+// RelayArgs represents [github.com/smartcontractkit/chainlink-relay/pkg/types.RelayArgs].
+message RelayArgs {
+  bytes externalJobID = 1; // [32]byte
+  int32 jobID = 2;
+  string contractID = 3;
+  bool new = 4;
+  bytes relayConfig = 5;
+}
+
+// RelayArgs represents [github.com/smartcontractkit/chainlink-relay/pkg/types.PluginArgs].
+message PluginArgs {
+  string transmitterID = 1;
+  bytes pluginConfig = 2;
+}
+
+// NewConfigProviderRequest has arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.NewConfigProvider].
+message NewConfigProviderRequest {
+  RelayArgs relayArgs = 1;
+}
+
+// NewConfigProviderReply has return arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.NewConfigProvider].
+message NewConfigProviderReply {
+  uint32 configProviderID = 1;
+}
+
+// NewMedianProviderRequest has arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.NewMedianProvider].
+message NewMedianProviderRequest {
+  RelayArgs relayArgs = 1;
+  PluginArgs pluginArgs = 2;
+}
+
+// NewMedianProviderReply has return arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.NewMedianProvider].
+message NewMedianProviderReply {
+  uint32 medianProviderID = 1;
+}
+
+// NewMercuryProviderRequest has arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.NewMercuryProvider].
+message NewMercuryProviderRequest {
+  RelayArgs relayArgs = 1;
+  PluginArgs pluginArgs = 2;
+}
+
+// NewMercuryProviderReply has return arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.NewMercuryProvider].
+message NewMercuryProviderReply {
+  uint32 mercuryProviderID = 1;
+}
+
+// ChainStatusRequest has arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.ChainStatus].
+message ChainStatusRequest {
+  string id = 1;
+}
+
+// ChainStatusReply has return arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.ChainStatus].
+message ChainStatusReply {
+  ChainStatus chain = 1;
+}
+
+// ChainStatusesRequest has arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.ChainStatuses].
+message ChainStatusesRequest {
+  int32 offset = 1;
+  int32 limit = 2;
+}
+
+// ChainStatusesReply has return arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.ChainStatuses].
+message ChainStatusesReply {
+  repeated ChainStatus chains = 1;
+  int32 count = 2; // total count of chains
+}
+
+// ChainStatus represents [github.com/smartcontractkit/chainlink-relay/pkg/types.ChainStatus].
+message ChainStatus {
+  string id = 1;
+  bool enabled = 2;
+  string config = 3; // TOML
+}
+
+// NodeStatusesRequest has arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.NodeStatuses].
+message NodeStatusesRequest {
+  int32 offset = 1;
```

Comment:
offset and limit should both be uintXX.
Or does the proto guidelines prefer all ints be int32?

Reply:
This was matching the existing types, but IIRC they also make use of `-1`.